### ISSUE TITLE
SSH producer test frequently swallows the output

### ISSUE THREAD

**Issue by tdiesler** [wildfly-extras/wildfly-camel#1564]
```
Failed tests: 
  SSHIntegrationTest.testSSHProducer:94 expected:<[Hello Kermit
]> but was:<[]>
```

**Comment by jamesnetherton** [comment]
I can't replicate this outside of the CI environment where it fails frequently. My assumption is that it's a race condition somewhere in the Apache SSHD client or server components. But it's proving really hard to track down.

It's very odd that no content from the SSH command seems to be returned. I've got byteman injecting sysouts into the Camel and Apache SSHD internals at runtime and this also seems to show that the OutputStream coming back from the process is empty. 

Perhaps the SSH exec channel is closed prematurely.

**Comment by jamesnetherton** [comment]
I managed to use Byteman to get the stacktrace of a [swallowed exception](https://github.com/apache/mina-sshd/blob/sshd-0.14.0/sshd-core/src/main/java/org/apache/sshd/server/shell/InvertedShellWrapper.java#L153) that seems to be the cause of this:

```
java.io.IOException: Stream closed
	at java.lang.ProcessBuilder$NullOutputStream.write(ProcessBuilder.java:433)
	at java.io.OutputStream.write(OutputStream.java:116)
	at java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:82)
	at java.io.BufferedOutputStream.flush(BufferedOutputStream.java:140)
	at java.io.FilterOutputStream.flush(FilterOutputStream.java:140)
	at org.apache.sshd.server.shell.InvertedShellWrapper.pumpStream(InvertedShellWrapper.java:165)
	at org.apache.sshd.server.shell.InvertedShellWrapper.pumpStreams(InvertedShellWrapper.java:135)
	at org.apache.sshd.server.shell.InvertedShellWrapper$1.run(InvertedShellWrapper.java:116)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
``` 

Not sure why this happens yet.

**Comment by jamesnetherton** [comment]
Camel 2.20.x has upgraded to Apache SSHD 1.6.0 so I'm going to mark this as fixed on our 2.20 branch. We can always reopen if it starts failing in the CI environment.
